src/main/java/net/dv8tion/jda/core/MessageBuilder.java

```diff
@@ -366,18 +431,505 @@ public int getLength()
+     *
+     * If no such value of <i>k</i> exists, then {@code -1} is returned.
+     *
+     * @param target the substring to search for.
+     * @param fromIndex the index from which to start the search.
+     * @param endIndex the index at which to end the search.
+     * @return  the index of the last occurrence of the specified substring between
+     *          the specified indices or {@code -1} if there is no such occurrence.
+     */
+    public int lastIndexOf(CharSequence target, int fromIndex, int endIndex)
+    {
+        if (fromIndex < 0)
+            throw new IndexOutOfBoundsException("index out of range: " + fromIndex);
+        if (endIndex < 0)
+            throw new IndexOutOfBoundsException("index out of range: " + endIndex);
+        if (fromIndex > length())
+            throw new IndexOutOfBoundsException("fromIndex > length()");
+        if (fromIndex > endIndex)
+            throw new IndexOutOfBoundsException("fromIndex > endIndex");
+        
+        if (endIndex >= builder.length())
+        {
+            endIndex = builder.length() - 1;
+        }
+
+        int targetCount = target.length();
+        if (targetCount == 0)
+        {
+            return endIndex;
+        }
+
+        int rightIndex = endIndex - targetCount;
+
+        if (fromIndex > rightIndex)
+        {
+            fromIndex = rightIndex;
+        }
+
+        int strLastIndex = targetCount - 1;
+        char strLastChar = target.charAt(strLastIndex);
+
+        int min = fromIndex + targetCount - 1;
+
+        lastCharSearch: for (int i = endIndex; i >= min; i--)
+        {
+            if (builder.charAt(i) == strLastChar)
+            {
+                for (int j = strLastIndex - 1, k = 1; j >= 0; j--, k++)
+                {
+                    if (builder.charAt(i - k) != target.charAt(j))
+                    {
+                        continue lastCharSearch;
+                    }
+                }
+                return i - target.length() + 1;
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Creates a {@link java.util.Queue Queue} of {@link net.dv8tion.jda.core.entities.Message Message} objects from this MessageBuilder.
+     * <p>
+     * This method splits the content if it exceeds 2000 chars. The spltting behaviour can be customized using {@link SplitPolicy SplitPolicies}.
+     * The method will try the policies in the order they are passed to it.
+     * <p>
+     * This is not MarkDown safe. An easy workaround is to include Zero Witdh Spaces as predetermined breaking points to the message and only split on them.
+     * 
+     * @return the created {@link net.dv8tion.jda.core.entities.Message Messages}
+     */
+    public Queue<Message> buildAll(SplitPolicy... policy)
+    {
+        if (builder.length() == 0)
+            throw new UnsupportedOperationException("Cannot build a Message with no content. (You never added any content to the message)");
```

Comment:
forgot to change to `this.isEmpty()` because I made that method after the build() methods